x-pack/plugins/lists/server/routes/create_list_index_route.ts

```diff
@@ -33,33 +33,43 @@ export const createListIndexRoute = (router: ListsPluginRouter): void => {
         const listIndexExists = await lists.getListIndexExists();
         const listItemIndexExists = await lists.getListItemIndexExists();
 
+        try {
+          // Check if the old legacy lists and items template exists and remove it
+          await lists.deleteLegacyListBootStrapIndex();
+          await lists.deleteLegacyListItemBootStrapIndex();
+        } catch (err) {
+          if (err.statusCode !== 404) {
+            throw err;
+          }
+        }
+
+        const policyExists = await lists.getListPolicyExists();
+        const policyListItemExists = await lists.getListItemPolicyExists();
+
+        if (!policyExists) {
+          await lists.setListPolicy();
+        }
+        if (!policyListItemExists) {
+          await lists.setListItemPolicy();
+        }
+
+        const templateExists = await lists.getListTemplateExists();
+        const templateListItemsExists = await lists.getListItemTemplateExists();
+
+        if (!templateExists) {
+          await lists.setListTemplate();
+        }
+
+        if (!templateListItemsExists) {
+          await lists.setListItemTemplate();
+        }
+
         if (listIndexExists && listItemIndexExists) {
           return siemResponse.error({
             body: `index: "${lists.getListIndex()}" and "${lists.getListItemIndex()}" already exists`,
             statusCode: 409,
           });
         } else {
-          const policyExists = await lists.getListPolicyExists();
```

Comment:
It seems like moving the policy stuff earlier should be fine; was there a need for moving it, or was this a logical refactoring?

Reply:
Just a logical refactoring and matching the patterns in security_solution from their refactoring to use new template endpoints.